### PR TITLE
Write results to disk and to stdout

### LINKED ISSUE #14: Consider writing check results to a file
As a user, I may prefer for the results of my preflight execution to be written to a file instead of to STDOUT. Note that this refers to the test output that is passed through the implemented formatters itself, and not the log output the that preflight provides to the user at execution time.

**Implementation Note:**

We write our formatted text in one place: https://github.com/redhat-openshift-ecosystem/openshift-preflight/blob/ae9b2c7a5e78c092d574de7425392ce1a6faea89/cmd/root.go#L56 

This should just be a matter of choosing where to write.

### PR DESCRIPTION
**Pull request by komish**
Fixes #14 

## Summary

This PR configures preflight to write the check Results to a file as well as stdout using a MultiWriter.

This PR recognizes that different formatter types might have different file extensions, so this extends to ResponseFormatter interface to allow for the definition of the desired file extension [3ad9a1a22c757491d93d20dc345e70424732df4c](https://github.com/redhat-openshift-ecosystem/openshift-preflight/commit/3ad9a1a22c757491d93d20dc345e70424732df4c).

The writers then take this into account to write the file. [48140895b3ae9d6fa17a9da0f9ddcddd2082376b](https://github.com/redhat-openshift-ecosystem/openshift-preflight/commit/48140895b3ae9d6fa17a9da0f9ddcddd2082376b)

## Highlights
- Removed AllFormats() function as it was not in use.
- Made the formatters_test suite and internal test. This was previously an external test suite due to a dependency cycle with the cmd package which has now been removed. An equivalent test should live in the `cmd` package.
- Introduced a new interface method ResponseFormatter.FileExtension() which returns a string containing the file extension someone might use when writing the data from that formatter to disk.
- Expended the genericFormatter to continue implementing the ResponseFormatter interface
- Reduced the public surface of the genericFormatter
- Added comments to the interface definition.
- Added a function that takes the extension defined by the formatter and concatenates it with our expected results file.
- Modified the `check container` and `check operator` subcommands to write to a default file.

### CODE REVIEW

**Review by samira-barouti — APPROVED** [review]

File: cmd/check_container.go [hunk 1]
```diff
@@ -43,16 +44,46 @@ var checkContainerCmd = &cobra.Command{
...
 			return err
 		}
 
-		fmt.Fprint(os.Stdout, string(formattedResults))
+		fmt.Fprint(resultsOutputTarget, string(formattedResults))
+		if err := resultsFile.Close(); err != nil {
+			return err
+		}
+
+		// engine.ExecuteChecks()
```
Comment (bcrochet) [inline cmd/check_container.go:77]: Big block of commented code.

**Review by bcrochet — CHANGES_REQUESTED** [review]

File: cmd/check_container.go (hunk 1 above)
Comment (komish) [reply cmd/check_container.go:77]: 🤦 I'll fix and push tomorrow. Thanks for the catch!

Comment (komish) [reply cmd/check_container.go]: @bcrochet this should be fixed.

**Review by bcrochet — APPROVED** [review]
/lgtm

**Pull request merged**